crates/core/src/types/engine/payload.rs

```diff
@@ -73,11 +73,19 @@ impl EncodedTransaction {
                 match *tx_type {
                     // Legacy
                     0x0 => LegacyTransaction::decode(tx_bytes).map(Transaction::LegacyTransaction), // TODO: check if this is a real case scenario
+                    // EIP2930
+                    0x1 => {
+                        EIP2930Transaction::decode(tx_bytes).map(Transaction::EIP2930Transaction)
+                    }
                     // EIP1559
                     0x2 => {
                         EIP1559Transaction::decode(tx_bytes).map(Transaction::EIP1559Transaction)
                     }
-                    _ => unimplemented!("We don't know this tx type yet"),
+                    // EIP4844
+                    0x3 => {
+                        EIP4844Transaction::decode(tx_bytes).map(Transaction::EIP4844Transaction)
+                    }
+                    _ => Err(RLPDecodeError::InvalidTransactionType),
```

Comment:
Can we change the return type, or return `MalformedData` instead?

Reply:
Could we use the custom variant? It is a shame to lose key information useful for debugging

Reply:
Sure!